skyportal/handlers/api/photometry.py

```diff
@@ -493,323 +566,125 @@ def put(self):
+                              description: List of new photometry IDs
+                            upload_id:
+                              type: string
+                              description: |
+                                Upload ID associated with all photometry points
+                                added in request. Can be used to later delete all
+                                points in a single request.
+        """
 
-            existing_hashes = [d.hash for d in duplicated_photometry]
-
-            if len(existing_hashes) > 0:
-                # save existing ids:
-                existing_ids.extend([d.id for d in duplicated_photometry])
-
-                # update groups if necessary
-                for duplicate in duplicated_photometry:
-                    duplicate_group_ids = set([g.id for g in duplicate.groups])
-
-                    # posting to new groups?
-                    if len(set(group_ids) - duplicate_group_ids) > 0:
-                        # select old + new groups
-                        group_ids_update = set(group_ids).union(duplicate_group_ids)
-                        groups = (
-                            DBSession()
-                            .query(Group)
-                            .filter(Group.id.in_(group_ids_update))
-                            .all()
-                        )
-                        # update the corresponding photometry entry in the db
-                        duplicate.groups = groups
+        try:
+            group_ids = self.get_group_ids()
+        except ValidationError as e:
+            return self.error(e.args[0])
 
-                w_oid_duplicate = (df.obj_id == oid) & (df.hash.isin(existing_hashes))
-                # now safely drop the duplicates:
-                df = df.drop(index=np.where(w_oid_duplicate)[0])
+        try:
+            df, instcache = self.standardize_photometry_data()
+        except ValidationError as e:
+            return self.error(e.args[0])
 
-        # posted data contains only duplicates already present in the db?
-        if len(df) == 0:
-            DBSession().commit()
-            return self.success(data={"ids": existing_ids})
+        values_table, condition = self.get_values_table_and_condition(df)
 
-        # pre-fetch the photometry PKs. these are not guaranteed to be
-        # gapless (e.g., 1, 2, 3, 4, 5, ...) but they are guaranteed
-        # to be unique in the table and thus can be used to "reserve"
-        # PK slots for uninserted rows
-        pkq = (
-            f"SELECT nextval('photometry_id_seq') FROM "
-            f"generate_series(1, {len(df)})"
+        new_photometry_query = (
+            DBSession()
+            .query(values_table.c.pdidx)
+            .outerjoin(Photometry, condition)
+            .filter(Photometry.id.is_(None))
         )
 
-        proxy = DBSession().execute(pkq)
+        new_photometry_ids = [g[0] for g in new_photometry_query]
```

Comment:
I didn't realize you could iterate through the query like this instead of the list returned by `new_photometry_query.all()`. Is there a reason for doing this instead of having the .all() at the end of the query above (and like you have for the duplicated version of this right below)?

Reply:
Just for debugging purposes. Helps to be able to have a reference to the query which I can then print.

Reply:
That doesn't seem to warrant leaving it in a non-standard (at least throughout our codebase) format; it's easily changed if you need to debug and add a print statement. I'd suggest using `.all()` for consistency

Reply:
Also, if these values are indices rather than DB IDs, then the variable name should be changed accordingly (e.g. ids -> df_idxs or (even better) df_indices)

Reply:
@acrellin renamed

Reply:
@acrellin No need for `all` here as we are not returning an instance of a mapped class from the query, just a single value (`id`), which would need its own iterator for extraction after `.all()` anyway, as the query returns it as a list of 1-element tuples